rosbag2_test_common/include/rosbag2_test_common/client_manager.hpp

```diff
@@ -0,0 +1,126 @@
+#define ROSBAG2_TEST_COMMON__CLIENT_MANAGER_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rcl/service_introspection.h"
+
+#include "rclcpp/rclcpp.hpp"  // rclcpp must be included before the Windows specific includes.
+
+
+namespace rosbag2_test_common
+{
+template<typename ServiceT>
+class ClientManager : public rclcpp::Node
+{
+public:
+  explicit ClientManager(
+    std::string service_name,
+    size_t client_number = 1,
+    bool service_event_contents = false,
+    bool client_event_contents = true)
+  : Node("service_client_manager_" + std::to_string(rclcpp::Clock().now().nanoseconds()),
+      rclcpp::NodeOptions().start_parameter_services(false).start_parameter_event_publisher(
+        false).enable_rosout(false)),
+    service_name_(service_name),
+    client_number_(client_number),
+    enable_service_event_contents_(service_event_contents),
+    enable_client_event_contents_(client_event_contents)
+  {
+    auto echo_process =
+      [this](const std::shared_ptr<rmw_request_id_t> request_header,
+        const std::shared_ptr<typename ServiceT::Request> request,
+        std::shared_ptr<typename ServiceT::Response> response) -> void
+      {
+        // Do nothing
+        (void)request_header;
+        (void)request;
+        (void)response;
+      };
+
+    service_ = create_service<ServiceT>(service_name_, echo_process);
+
+    rcl_service_introspection_state_t introspection_state;
+    if (enable_service_event_contents_) {
+      introspection_state = RCL_SERVICE_INTROSPECTION_CONTENTS;
+    } else {
+      introspection_state = RCL_SERVICE_INTROSPECTION_OFF;
+    }
+    service_->configure_introspection(
+      get_clock(), rclcpp::SystemDefaultsQoS(), introspection_state);
+
+    if (enable_client_event_contents_) {
+      introspection_state = RCL_SERVICE_INTROSPECTION_CONTENTS;
+    } else {
+      introspection_state = RCL_SERVICE_INTROSPECTION_OFF;
+    }
+
+    for (size_t i = 0; i < client_number_; i++) {
+      auto client = create_client<ServiceT>(service_name_);
+      client->configure_introspection(
+        get_clock(), rclcpp::SystemDefaultsQoS(), introspection_state);
+      clients_.emplace_back(client);
+    }
+  }
+
+  bool check_service_ready()
+  {
+    for (auto & client : clients_) {
+      if (!client->service_is_ready()) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  bool send_request()
+  {
+    if (!check_service_ready()) {
+      return false;
+    }
+
+    for (auto & client : clients_) {
+      auto request = std::make_shared<typename ServiceT::Request>();
+      auto result = client->async_send_request(request);
+      // Wait for the result.
+      if (rclcpp::executors::spin_node_until_future_complete(
+          exec_, get_node_base_interface(), result) != rclcpp::FutureReturnCode::SUCCESS)
+      {
+        RCLCPP_INFO(
+          rclcpp::get_logger("service_client_manager"), "Failed to get response !");
```

Comment:
Comments from MichaelOrlov

```suggestion
          RCLCPP_INFO(this->get_logger(), "Failed to get response !");
```
Nitpick. To avoid occasional node name mismatch

Reply:
Upddated